### DISCUSSION TITLE
Options to write ElastAlert2 metadata

### DISCUSSION THREAD

**Discussion by GraFindlay** [jertel/elastalert2#807] [Q&A] [answered]
As an organisation we are not allowed to use ElastAlert2 to create/update an index.  This means that it likely that we cannot use some of the rules (I'm assuming ones that use aggregation etc).

Are there any other options available to store metadata either in a flat file or DB?  If there are no current options is there any ongoing work that is looking to address this?

Thanks.

**Comment by jertel** [comment, marked as answer]
ElastAlert 2 needs to have the ability to create and update it's own elastalert_* indices, for all rule types. There are no plans to support storage of that metadata outside of the Elasticsearch cluster. 

Elasticsearch provides the ability to limit write access on a per-index basis to specific Elasticsearch users. So your organization could grant write access for the elastalert user only to the elastalert_* indices but not to other indices.

**Reply by GraFindlay** [reply]
OK, thanks for the update.
